src/platform/lumin-runtime/native-file-system.js

```diff
@@ -1,19 +1,42 @@
+import { open, write, close, read } from "magic-script-polyfills/src-ts/fs-promised.ts"
+import { b64ToBin, rawToBin, strToBin, binToStr } from "magic-script-polyfills/src/bintools.js"
+
+
 class NativeFileSystem {
     
-    writeFile(path, content, encoding) {
-        return Promise.reject(new Error('Write file function is not implemented yet'));
+     async writeFile(path, contents, encoding) {
+        const fd = await open(path, "w", 0o644);
+        try {
+            const body = encoding === 'utf8' ? strToBin(contents)
+            : encoding === 'base64' ? b64ToBin(contents)
+            : encoding === 'ascii' ? rawToBin(contents) 
+            : `Unsupported encoding '${encoding}'`;
+            if(body === `Unsupported encoding '${encoding}'`) {
+                throw new Error(body)
+            }
+            await write(fd, body, 0);
+        } finally {
+            await close(fd);
+        }
     }
 
-    readFile(path, encoding) {
-        return Promise.reject(new Error('Read file function is not implemented yet'));
+    async readFile(path, encoding) {
+        const fd = await open(path, "r", 0o444);
+        try {
+            let content = new Uint8Array(64 * 1024);
```

Comment:
This needs to not hard-code the file length unless you know all files are exactly 64k long.